ports/student.py

```diff
@@ -0,0 +1,22 @@
+class Student:
+    def __init__(self, username: str, scores: list[float]):
+        self._username = username
+        self._scores = scores
+
+    def calculate_average(self) -> float:
```

Comment:
Please make sure that average is calculated for the first function invocation only; further invocations should return the previously calculated value (lazy evaluation)